### DISCUSSION TITLE
Scheduler really slow in marking `dag_run` successful when all tasks are skipped

### DISCUSSION THREAD

**Discussion by V0lantis** [apache/airflow#37448] [Q&A]
We noticed that our Airflow was really slow in scheduling and marking `dag_run` to success when dealing with only skipped tasks

We have couple of dags with 3 tasks in it. It is a very simple dag with only one branch. Tasks are run if data is present and to test that there is data, we have a sensor. If the sensor is skipped, then by propagation, all downstream tasks are skipped. But our Airflow is immensely slow in skipping the downstream tasks, and even more so in passing the dag in success.

### What do I expect

I would expect the step of skipping tasks and marking dag_run to success to be almost instantaneous

**Comment by Taragolis** [comment]
"Really slow" it is abstract definition and depend on personal feelings, e.g. for some one 0.1s would be slow, for the others 10s is fine and pretty fast.
Could you elaborate a bit more with provide examples, task durations and version of Airflow and DB (type and version)

**Comment by V0lantis** [comment]
> "Really slow" it is abstract definition and depend on personal feelings, e.g. for some one 0.1s would be slow, for the others 10s is fine and pretty fast.

That's a very good point. I am unable to upload an image, so Ill try to be as clear as possible:

The DAGs are automatically generated so they all have the same pattern. Just the data source and output changes. They all start with a sensor, which is, in the beginning skipped because there is no data. The scheduling and the skipped status is almost instantaneous (<1second). But then, the next two tasks which should be marked skipped as well, take around 5 minutes to be skipped, and it takes 5 more minutes to mark the DAG to success. So I am wondering where lie the issue

> Could you elaborate a bit more with provide examples, task durations and version of Airflow and DB (type and version)

Airflow version 2.7.3
postgresql 15.5
Running on Kubernetes, with CeleryKubernetesExecutor (and keda autoscalling)
Resources of the scheduler don't look overused
Arrounf ~4000 dag_run in running state

**Reply by potiuk** [reply]
I think without seeing logs and examples, this one is not actionable. Anything that you can provide: explaining your deployment, showing logs of scheduler doing scheduling decisions around the time when your slowdowns happens, extractng and publishing the logs for particular single DAG that you will be able to post together with correlated logs, is the only way you can help others with helping to investigate your problem, I am afraid.

**Reply by V0lantis** [reply]
Fair point. Ill try to do so then
